data/blog/session-replay-sdk-bundle-size-optimizations.md

```diff
@@ -0,0 +1,209 @@
+images: [/images/session-replay-sdk-bundle-size-optimizations/hero.jpg]
+layout: PostLayout
+canonicalUrl: https://sentry.engineering/blog/session-replay-sdk-bundle-size-optimizations
+authors: ['francesconovy']
+---
+
+[Bundle Size matters](https://blog.sentry.io/js-browser-sdk-bundle-size-matters/) - this is something we SDK engineers at Sentry are acutely aware of.
+In an ideal world, you'd get all the functionality you want with no additional bundle size - oh, wouldn't that be nice?
+Sadly, in reality any feature we add to the JavaScript SDK results in additional bundle size for the SDK - there is always a trade off to be made.
+
+With [Session Replay](https://docs.sentry.io/product/session-replay/), this topic is especially challenging.
+Session Replay allows to capture what's going on in a users browsers, which can help developers debug errors or problems the user is experiencing.
+While this can be incredibly helpful, there is also a considerable amount of JavaScript code required to actually make this possible - thus leading to an increased bundle size.
+
+In version 7.73.0 of the JavaScript SDKs, we updated the underlying [rrweb](https://github.com/getsentry/rrweb) package from v1 to v2.
+While this brought a host of improvements, it also came with a considerable increase in bundle size.
+This tipped us over the edge to declare a bundle size emergency, and focus on bringing the additional size Session Replay adds to the SDK down as much as possible.
+
+We're very happy to say that our efforts have been successful, and we managed to reduce the minified & gzipped bundle size compared to the rrweb 2 baseline by 23% (~19 KB), and by up to 35% (~29 KB) with maximum tree shaking configuration enabled.
+
+| Version                                                                      | Bundle Size¹ | What                                     |
+| ---------------------------------------------------------------------------- | ------------ | ---------------------------------------- |
+| [7.72.0](https://github.com/getsentry/sentry-javascript/releases/tag/7.72.0) | 75.58 KB     | Before updating to rrweb 2.0             |
+| [7.73.0](https://github.com/getsentry/sentry-javascript/releases/tag/7.73.0) | 84.26 KB     | After updating to rrweb 2.0              |
+| [7.78.0](https://github.com/getsentry/sentry-javascript/releases/tag/7.78.0) | 65.24 KB     | New default                              |
+| [7.78.0](https://github.com/getsentry/sentry-javascript/releases/tag/7.78.0) | 55.48 KB     | With all tree shaking options configured |
+
+¹Including Error & Performance Monitoring as well as Session Replay, minified & gzipped
+
+## Steps to reduce the SDK bundle size
+
+In order to achieve these bundle size improvements, we took a couple of steps ranging from removing unused code to build time configuration and improved tree shaking:
+
+- Allow to remove iframe & shadow DOM support via a build-time flag
```

Comment:
Okay so I notice this is in multiple parts of the blog post - "Allow to remove..." - it sounds awkward/not sure if grammatically correct (?). Maybe we can say "Allow you to..." or "Allows for removing..." or "Ability to remove..."

Reply:
WDYT about `Make it possible to remove...`?